examples/full_featured/plotly-example-full.agent.nut

```diff
@@ -0,0 +1,275 @@
+                "name" : trace
+            });
+            _persistentStyle.append({
+                "name" : trace  
+            });
+        }
+
+        _makeApiCall(MESSAGETYPE_PLOT, plotlyInput, callback);
+    }
+    
+    function getUrl() {
+        return _url;
+    }
+    
+    function post(dataObjs, callback = null) {;
+        _makeApiCall(MESSAGETYPE_PLOT, dataObjs, callback);
+    }
+    
+    function setTitle(title, callback = null) {
+        _persistentLayout["title"] <- title;
+        _makeApiCall(MESSAGETYPE_LAYOUT, _persistentLayout, callback);
+    }
+    
+    function setAxisTitles(xAxisTitle, yAxisTitle, callback = null) {
+        if(xAxisTitle != null && xAxisTitle.len() > 0) {
+            _persistentLayout["xaxis"]["title"] <- xAxisTitle;
+        }
+        if(yAxisTitle != null && yAxisTitle.len() > 0) {
+            _persistentLayout["yaxis"]["title"] <- yAxisTitle;
+        }
+        _makeApiCall(MESSAGETYPE_LAYOUT, _persistentLayout, callback);
+    }
+    
+    function addSecondYAxis(axisTitle, traces, callback = null) {
+            _persistentLayout["yaxis2"] <- {
+                "title" : axisTitle,
+                "side" : "right",
+                "overlaying" : "y"
+            };
+            // Search for requested traces in style table
+            foreach(trace in _persistentStyle) {
+                if(traces.find(trace["name"]) != null) {
+                    trace["yaxis"] <- "y2";
+                }
+            }
+            local secondAxisCallback = _getSecondAxisLayoutCallback(callback).bindenv(this);
+            _makeApiCall(MESSAGETYPE_LAYOUT, _persistentLayout, secondAxisCallback);
+    }
+    
+    function setStyleDirectly(styleTable, callback = null) {
+        // Note that this overwrites the existing style table
+        _persistentStyle = styleTable;
+        _makeApiCall(MESSAGETYPE_STYLE, _persistentStyle, callback);
+    }
+    
+    function setLayoutDirectly(layoutTable, callback = null) {
+        // Note that this overwrites the existing layout table
+        _persistentLayout = layoutTable;
+        _makeApiCall(MESSAGETYPE_LAYOUT, _persistentLayout, callback);
+    }
+    
+
+    /******************** PRIVATE FUNCTIONS (DO NOT CALL) ********************/
+    function _getSecondAxisStyleCallback(err1, response1, parsed1, userCallback) {
+        return function(err2, response2, parsed2) {
+            if(userCallback != null) {
+                // Since adding a second y-axis requires two API calls, pass the "worse" response into the user callback
+                local returnedResponse = response1.statuscode > response2.statuscode ? response1 : response2;
+                local returnedErr = response1.statuscode > response2.statuscode ? err1 : err2;
+                local returnedParsed = response1.statuscode > response2.statuscode ? parsed1 : parsed2;
+                imp.wakeup(0, @() userCallback(returnedErr, returnedResponse, returnedParsed));
```

Comment:
Should probably change `@() userCallback(returnedErr, returnedResponse, returnedParsed)` to `function() { userCallback(returnedErr, returnedResponse, returnedParsed); }` to match the common style of inline anonymous functions